frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/AttachmentStackedBar.tsx

```diff
@@ -28,65 +31,99 @@ const FilledTrack = (props: BoxProps) => {
 }
 
 export const AttachmentStackedBar = ({
+  existingValues: existingValuesProp,
+  newValue = 0,
   min = 0,
   max,
-  values,
 }: AttachmentStackedBarProps): JSX.Element => {
+  const isLoading = useMemo(() => !existingValuesProp, [existingValuesProp])
+
+  const existingValues = useMemo(
+    () => existingValuesProp ?? [],
+    [existingValuesProp],
+  )
+
+  const totalAttachmentSize = useMemo(
+    () => existingValues.reduce((sum, v) => sum + v, 0) + newValue,
+    [existingValues, newValue],
+  )
+
   const isOverQuota = useMemo(
-    () => !!values && max < values.reduce((sum, v) => sum + v, 0),
-    [max, values],
+    () => !!existingValues && max < totalAttachmentSize,
+    [max, totalAttachmentSize, existingValues],
   )
 
   const barProps = useMemo(() => {
-    return [
-      { bg: 'warning.500' },
-      {
-        bg: isOverQuota ? 'danger.500' : 'success.500',
-        border: isOverQuota
-          ? undefined
-          : '1px dashed var(--chakra-colors-success-800)',
-      },
-    ]
-  }, [isOverQuota])
+    const barProps = existingValues.map(() => ({
+      bg: 'warning.500',
+      border: 'none',
+    }))
+    if (newValue > 0)
+      barProps.push({
+        bg: 'success.500',
+        border: '1px dashed var(--chakra-colors-success-800)',
+      })
+    return barProps
+  }, [existingValues, newValue])
 
   const gridTemplateColumns = useMemo(() => {
-    return [
-      `${valueToPercent(values?.[0] ?? 0, min, max)}%`,
-      `minmax(auto, ${valueToPercent(values?.[1] ?? 0, min, max)}%)`,
-    ].join(' ')
-  }, [max, min, values])
+    const gridTemplateColumns = existingValues.map(
+      (value) => `${valueToPercent(value, min, max)}%`,
+    )
+    if (newValue > 0)
+      gridTemplateColumns.push(
+        `minmax(auto, ${valueToPercent(newValue, min, max)}%)`,
+      )
```

Comment:
nit: wrap with braces for readability?

Reply:
added in [270f849](https://github.com/opengovsg/FormSG/pull/4565/commits/270f849d96cc309b531c5688e1f0464de2f3ab91)